remoteworkitem/trackeritem_repository.go

```diff
@@ -5,6 +5,14 @@ import "github.com/jinzhu/gorm"
 // upload imports the items into database
 func upload(db *gorm.DB, tqID int, item map[string]string) error {
 	ti := TrackerItem{Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
-	err := db.Create(&ti).Error
+	db.Where("remote_item_id = ?", item["id"]).Find(&ti)
+	var err error
+	if ti.ID == 0 {
+		ti1 := TrackerItem{Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
+		err = db.Create(&ti1).Error
+	} else {
+		ti2 := TrackerItem{ID: ti.ID, Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
```

Comment:
My bad, yes it will save but that's not want we wanted.
My test cases fails, if I do that: https://gist.github.com/baijum/630f8d865c1a4ada93d10a01906255c1

Since we are calling `Find`, the existing data will be fetched and updated in the struct. But during update, we want the new data.